dp/generate_climos.py

```diff
@@ -583,6 +608,79 @@ def update_metadata_and_time_var(input_file, t_start, t_end, operation, climo_fi
 
     return climo_filepath
 
+def update_generate_climos_history(input_filepath, outdir, operation, convert_longitudes, 
+                                    split_vars, split_intervals, output_resolutions, netCDF_file, time_cdo_format, position=0):
+    """This function takes the start time and end time history-attribute-lines
+    of generate_climos command. It returns a string of the entire history
+    attribute. The result indicates when the generate_climos started and ended.
+
+    :param netCDF_file: (CFDataset) netCDF file to be updated
+    :param time_cdo_format: (str) the point of time that generate_climos history(start or end) was updated
+    :param position: (int) the position that the new history attribute line will be inserted
+    
+    ----------the following command line arguments will be included in the updated history attribute----------
+    :param input_filepath(str), outdir(str), operation(str), convert_longitudes(boolean), split_vars(boolean),
+            split_intervals(boolean), output_resolutions(set)
+    ----------------------------------------------------------------------------------------------------------
+
+    For example:
+
+    :history = 
+        "Thu May 21 11:12:04 2020: cdo -O -seldate ..."    <--- position 0
+        "Thu May 21 11:12:04: start: generate_climos ..."  <--- position 1: start of genereate_climos
+        "Thu Mar 21 14:49:01 2019: cdo sellonlatbox ..."   <--- position 2
+        "Thu Sep  1 14:34:03 2016: ncrcat ..."             <--- position 3
+        "Thu Sep 01 14:33:15 2016: cdo -O seldate ..."     <--- position 4
+        
+        ...
+
+    or
+
+    :history = 
+        "Thu May 21 11:12:05: end: generate_climos ..."    <--- position 0: end of genereate_climos
+        "Thu May 21 11:12:05 2020: cdo -O -replace ..."    <--- position 1 
+        "Thu May 21 11:12:04 2020: cdo -O -timmean ..."    <--- position 2
+        "Thu May 21 11:12:04 2020: cdo -O -seldate ..."    <--- position 3
+        "Thu May 21 11:12:04: start: generate_climos ..."  <--- position 4
+        "Thu Mar 21 14:49:01 2019: cdo sellonlatbox ..."   <--- position 5
+        "Thu Sep  1 14:34:03 2016: ncrcat ..."             <--- position 6
+        "Thu Sep 01 14:33:15 2016: cdo -O seldate ..."     <--- position 7
+
+        ...
+    
+    """
+    with CFDataset(netCDF_file, "r+") as cf:
+        history = cf.history
+        command = (": generate_climos" + " -p " + operation + " -o " + outdir + " " + input_filepath)
+
+        if not convert_longitudes:
+            command += " -g " + str(convert_longitudes)
+
+        if not split_vars:
+            command += " -v " + str(split_vars)
+
+        if not split_intervals:
+            command += + " -i " + str(split_intervals)
+
+        if output_resolutions != set(["yearly", "seasonal", "monthly"]):
+            command += " -r " + str(output_resolutions).split("\'")[1]
```

Comment:
I wonder if it would be better to use `sys.argv` to retrieve the command line arguments as entered rather than reconstructing them like this. It would certainly be less work. It would echo the actual original command line, rather than the defaults supplied. What are your thoughts?

Reply:
:+1: for using `sys.argv`!